style/crawler/crawl.py

```diff
@@ -1,29 +1,148 @@
-from bs4 import BeautifulSoup
+# from bs4 import BeautifulSoup
+from abc import ABC
+from abc import abstractmethod
+from os import mkdir
+from os import path
+from pathlib import Path
+
 import requests
 
+from utils.author_catalog import create_catalog
+from utils.author_catalog import read_catalog
+from utils.utils import sanitize_name
+
+gutenberg_base_url = "http://aleph.gutenberg.org/"
+file_path_book_db = path.join(Path(__file__).parents[2], "book_db")
 
-class Crawler:
+
+class DataWrangler:
     def __init__(self, baseurl):
-        self.baseurl = baseurl
         self.session = requests.Session()
+        self.baseurl = baseurl
 
+    @abstractmethod
     def crawl(self):
-        pass
+        raise NotImplementedError
 
+    @abstractmethod
     def parse(self):
-        pass
-# https://www.mustgo.com/worldlanguages/languages-a-z/
+        raise NotImplementedError
 
-class GutenbergCrawler(Crawler):
-    # Simdi ilk once bizim sectigimiz yazar listesine bakacak idsini verecegiz
-    #  id'den yazarin kitap linklerini tiklayacak
-    #  ararken kelielerin icinde german vs geciyorsa onlari atlayacak
-    #  burada dile gore filtreleme  (lowercase check etme)
-    #  kitaplara girdiginde orada da utf8'i linkini bulup
-    #  txt dosyasini kaydedecek (author-book-name)
 
+class GutenbergWrangler(DataWrangler, ABC):
+    def __init__(self, baseurl=gutenberg_base_url):
+        super().__init__(baseurl)
 
-    pass
+    def get_book_full_url(self, book_id: str) -> str:
+        book_id = str(book_id).strip()
+        if len(book_id) == 1:
+            url = f"0/{book_id}/{book_id}.txt"
+        elif len(book_id) == 2:
+            url = f"{book_id[0]}/{book_id}/{book_id}.txt"
+        elif len(book_id) == 3:
+            url = f"{book_id[0]}/{book_id[1]}/{book_id}/{book_id}.txt"
+        elif len(book_id) == 4:
+            url = f"{book_id[0]}/{book_id[1]}/{book_id[2]}/{book_id}/{book_id}.txt"
+        elif len(book_id) == 5:
+            url = f"{book_id[0]}/{book_id[1]}/{book_id[2]}/{book_id[3]}/{book_id}/{book_id}.txt"
+        else:
+            raise ValueError(
+                "Book's ID should be number between 1 and 65538. It was given {book_id}."
+            )
+        return path.join(self.baseurl, url)
+
+    def get_book(self, book_id):
+        """
+        It takes the book's url and returns the book's text.
+        Todo:
```

Comment:
this is good but do not remove the `Args`. Also fix the comments.

Reply:
I haven't removed the comments yet.